interface/src/app/analyze/analysis/analysis.js

```diff
@@ -187,21 +221,20 @@ AnnotationType, SampleBin) {
     ]
   };
 
-  // TODO these exampleCols are temporarily hard-coded until a column chooser
-  // feature can be added
-  $scope.analysis = {
-    exampleCols: [
-      {'typename': 'genotype'},
-      {'typename': 'medium'},
-      {'typename': 'strain'}
-    ]
-  };
-
   // populate sample details
   // FIXME implement this loop as a method on SampleBin? (prob. with caching)
+  $scope.analysis.status = 'Retrieving sample details';
+  var pArrSamples = [];
   for (var i = 0; i < SampleBin.heatmapData.samples.length; i++) {
-    SampleBin.getSampleDetails(SampleBin.heatmapData.samples[i]);
+    pArrSamples.push(
+      SampleBin.getSampleDetails(SampleBin.heatmapData.samples[i])
+    );
   }
+  $q.all(pArrSamples).then(function() {
```

Comment:
I thought about this for a while. Retrieving samples independently allows us to do caching at the level of samples. This means that if a user adds or removes a sample from the heatmap we can rebuild it much more quickly than we could otherwise.

In addition, although I haven't implemented it yet, we can ultimately begin fetching the analysis data into the cache as soon as an individual sample is added to the `sampleBin` (rather than waiting until the `analysis` view is opened, as we do now). This will make the apparent performance of the UI even snappier.

No need for a new API, by the way. The `Activity` API can already filter on `sample_id__in`. Before I implemented the cache this is the way it was done, in fact. Without caching, it *is* more efficient to get all of the activity data in a single call, but with the cache, I think this way is better.

Does that approach make sense?